### PR TITLE
[REVIEW ONLY] JAMES-2578 Fix warnings use Mailet new API in james-server-mailets

### PR DESCRIPTION
**Pull request by blackheaven**
See #1883

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
[d6ca7c4e0cb564925042deba6f678462dd89defe] [INFO] -------------------------------------------------------------
[d6ca7c4e0cb564925042deba6f678462dd89defe] [ERROR] COMPILATION ERROR : 
[d6ca7c4e0cb564925042deba6f678462dd89defe] [INFO] -------------------------------------------------------------
[d6ca7c4e0cb564925042deba6f678462dd89defe] [ERROR] /james-project/server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/WithPriority.java:[65,60] cannot find symbol
[d6ca7c4e0cb564925042deba6f678462dd89defe]   symbol:   variable MAIL_PRIORITY_ATTRIBUTE_NAME
[d6ca7c4e0cb564925042deba6f678462dd89defe]   location: interface org.apache.james.queue.api.MailPrioritySupport
[d6ca7c4e0cb564925042deba6f678462dd89defe] [ERROR] /james-project/server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/IsMarkedAsSpam.java:[76,100] cannot find symbol
[d6ca7c4e0cb564925042deba6f678462dd89defe]   symbol:   method asString()
[d6ca7c4e0cb564925042deba6f678462dd89defe]   location: variable STATUS_MAIL_ATTRIBUTE_NAME of type java.lang.String
[d6ca7c4e0cb564925042deba6f678462dd89defe] [ERROR] /james-project/server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/managesieve/ManageSieveMailet.java:[135,66] cannot find symbol
[d6ca7c4e0cb564925042deba6f678462dd89defe]   symbol:   method isPresent()
[d6ca7c4e0cb564925042deba6f678462dd89defe]   location: interface java.io.Serializable
[d6ca7c4e0cb564925042deba6f678462dd89defe] [ERROR] /james-project/server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remote/delivery/DeliveryRunnable.java:[171,64] cannot find symbol
[d6ca7c4e0cb564925042deba6f678462dd89defe]   symbol:   variable MAIL_PRIORITY_ATTRIBUTE_NAME
[d6ca7c4e0cb564925042deba6f678462dd89defe]   location: interface org.apache.james.queue.api.MailPrioritySupport
[d6ca7c4e0cb564925042deba6f678462dd89defe] [ERROR] /james-project/server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remote/delivery/DeliveryRunnable.java:[171,114] cannot find symbol
[d6ca7c4e0cb564925042deba6f678462dd89defe]   symbol:   variable LOW_PRIORITY_ATTRIBUTE_VALUE
[d6ca7c4e0cb564925042deba6f678462dd89defe]   location: interface org.apache.james.queue.api.MailPrioritySupport
[d6ca7c4e0cb564925042deba6f678462dd89defe] [ERROR] /james-project/server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java:[199,64] cannot find symbol
[d6ca7c4e0cb564925042deba6f678462dd89defe]   symbol:   variable MAIL_PRIORITY_ATTRIBUTE_NAME
[d6ca7c4e0cb564925042deba6f678462dd89defe]   location: interface org.apache.james.queue.api.MailPrioritySupport
[d6ca7c4e0cb564925042deba6f678462dd89defe] [INFO] 
[d6ca7c4e0cb564925042deba6f678462dd89defe] [ERROR] /james-project/server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java:[199,114] cannot find symbol
[d6ca7c4e0cb564925042deba6f678462dd89defe]   symbol:   variable HIGH_PRIORITY_ATTRIBUTE_VALUE
[d6ca7c4e0cb564925042deba6f678462dd89defe]   location: interface org.apache.james.queue.api.MailPrioritySupport
[d6ca7c4e0cb564925042deba6f678462dd89defe] [INFO] --- swagger-maven-plugin:3.1.7:generate (default) @ james-server-webadmin-data ---
[d6ca7c4e0cb564925042deba6f678462dd89defe] [ERROR] /james-project/server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/SpamAssassin.java:[113,71] incompatible types: java.lang.String cannot be converted to org.apache.mailet.AttributeName
[d6ca7c4e0cb564925042deba6f678462dd89defe] [ERROR] /james-project/server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/SpamAssassin.java:[116,67] cannot find symbol
[d6ca7c4e0cb564925042deba6f678462dd89defe]   symbol:   method getValue()
[d6ca7c4e0cb564925042deba6f678462dd89defe]   location: class java.lang.String
[d6ca7c4e0cb564925042deba6f678462dd89defe] [INFO] 9 errors 
```

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remote/delivery/DeliveryRetriesHelper.java [hunk 1]
```diff
@@ -19,32 +19,28 @@
...
-            if (value != null) {
-                return (Integer) value;
-            }
-            return 0;
-        } catch (ClassCastException e) {
-            return 0;
-        }
+        return AttributeUtils
+                .getValueAndCastFromMail(mail, DELIVERY_RETRY_COUNT, Integer.class)
+                .orElse(0);
```
Comment (aduprat) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remote/delivery/DeliveryRetriesHelper.java]: Maybe the time to introduce a constant for that:
`NONE_YET = 0`

**Comment by hoangdat** [comment]
Build started sha1 is merged.

**Comment by hoangdat** [comment]
Build finished. No test results found.

**Comment by hoangdat** [comment]
Build triggered. sha1 is merged.

**Pull request closed without merging**